[CI] ClientYamlTestSuiteIT test {yaml=simulate.ingest/10_basic/Test mapping validation from templates} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #3953 / rocky-8_platform-support-unix](https://gradle-enterprise.elastic.co/s/xspcxonjqpi4q)
- [elasticsearch-pull-request #30243 / part-1](https://gradle-enterprise.elastic.co/s/cbpheftyamfse)

**Reproduction Line:**
```
./gradlew ":rest-api-spec:yamlRestTest" --tests "org.elasticsearch.test.rest.ClientYamlTestSuiteIT" -Dtests.method="test {yaml=simulate.ingest/10_basic/Test mapping validation from templates}" -Dtests.seed=8ABC13F02418048C -Dtests.locale=se-SE -Dtests.timezone=CNT -Druntime.java=22
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupInput:(chainingSystem:HIERARCHICAL,controlStyle:twoLine,ignoreParentSettings:(ignoreFilters:!f,ignoreQuery:!f,ignoreTimerange:!f,ignoreValidations:!t),panels:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:task.keyword,grow:!t,id:'0c0c9cb8-ccd2-45c6-9b13-96bac4abc542',searchTechnique:wildcard,selectedOptions:!(),singleSelect:!t,title:'Gradle%20Task',width:medium),grow:!t,order:0,type:optionsListControl,width:small),'144933da-5c1b-4257-a969-7f43455a7901':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:name.keyword,grow:!t,id:'144933da-5c1b-4257-a969-7f43455a7901',searchTechnique:wildcard,selectedOptions:!('test%20%7Byaml%3Dsimulate.ingest%2F10_basic%2FTest%20mapping%20validation%20from%20templates%7D'),title:Test,width:medium),grow:!t,order:2,type:optionsListControl,width:medium),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(explicitInput:(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,enhancements:(),fieldName:className.keyword,grow:!t,id:'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850',searchTechnique:wildcard,selectedOptions:!('org.elasticsearch.test.rest.ClientYamlTestSuiteIT'),title:Suite,width:medium),grow:!t,order:1,type:optionsListControl,width:medium)))))

**Failure Message:**
```
java.lang.AssertionError: Failure at [simulate.ingest/10_basic:281]: got unexpected warning header [
	299 Elasticsearch-8.16.0-137094fcfff81d92d222079b21948d6d3feb055d "index template [v2_template] has index patterns [v2_strict_nonexistent*] matching patterns from existing older templates [global] with patterns (global => [*]); this template [v2_template] will take precedence during new index creation"
]

```

**Issue Reasons:**
- [main] 2 failures in test test {yaml=simulate.ingest/10_basic/Test mapping validation from templates} (1.1% fail rate in 176 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -178,9 +178,6 @@ tests:
 - class: org.elasticsearch.xpack.esql.EsqlAsyncSecurityIT
   method: testIndexPatternErrorMessageComparison_ESQL_SearchDSL
   issue: https://github.com/elastic/elasticsearch/issues/112630
-- class: org.elasticsearch.test.rest.ClientYamlTestSuiteIT
-  method: test {yaml=simulate.ingest/10_basic/Test mapping validation from templates}
-  issue: https://github.com/elastic/elasticsearch/issues/112633
 - class: org.elasticsearch.compute.aggregation.blockhash.BlockHashTests
   method: testBytesRefLongHashHugeCombinatorialExplosion {forcePackedHash=false}
   issue: https://github.com/elastic/elasticsearch/issues/112442

